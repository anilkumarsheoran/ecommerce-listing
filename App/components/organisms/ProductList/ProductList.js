import PropTypes from "prop-types";
import Product from "../../molecules/Product";
import withStyle from "../../../global/withStyle";
import style from "./ProductList.style";

/**
 * ProductList component can be used to display a listy of Products. This component is connected with Styled component
 *  to add style
 *  @param {object} items : items object contains array of object of prices, productNames and productType
 */

const ProductList = props => {
  const { items, className } = props;
  return (
    <ul className={className}>
      {items.map(item => (
        <Product item={item} key={item.index} />
      ))}
    </ul>
  );
};

ProductList.propTypes = {
  items: PropTypes.array.isRequired,
  className: PropTypes.string
};

export default withStyle(ProductList, style);
