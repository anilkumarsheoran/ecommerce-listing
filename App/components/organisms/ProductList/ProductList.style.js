import { css } from "styled-components";

export default `
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin-bottom: 30px;
`;
