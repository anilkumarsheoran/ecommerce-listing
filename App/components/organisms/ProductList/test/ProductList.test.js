import { shallow } from "enzyme";
import ProductList from "../ProductList";
const { configure } = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

configure({ adapter: new Adapter() });

const props = {
  items: [
    {
      index: 0,
      isSale: true,
      isExclusive: false,
      price: "$18.88",
      productImage: "product-1.jpg",
      productName: "Striped shirt",
      size: ["S", "L", "XL"]
    },
    {
      index: 1,
      isSale: false,
      isExclusive: false,
      price: "$25.44",
      productImage: "product-2.jpg",
      productName: "Denim shirt",
      size: ["XS", "S"]
    }
  ]
};

describe("ProductList Component", () => {
  test("should render correctly", () => {
    const productlist = shallow(<ProductList {...props} />);
    expect(productlist).toMatchSnapshot();
  });
});
