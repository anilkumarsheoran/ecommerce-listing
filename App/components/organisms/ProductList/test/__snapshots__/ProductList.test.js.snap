// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(ProductList)
    items={
      Array [
        Object {
          "index": 0,
          "isExclusive": false,
          "isSale": true,
          "price": "$18.88",
          "productImage": "product-1.jpg",
          "productName": "Striped shirt",
          "size": Array [
            "S",
            "L",
            "XL",
          ],
        },
        Object {
          "index": 1,
          "isExclusive": false,
          "isSale": false,
          "price": "$25.44",
          "productImage": "product-2.jpg",
          "productName": "Denim shirt",
          "size": Array [
            "XS",
            "S",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "className": "sc-htpNat kDFENX",
      "items": Array [
        Object {
          "index": 0,
          "isExclusive": false,
          "isSale": true,
          "price": "$18.88",
          "productImage": "product-1.jpg",
          "productName": "Striped shirt",
          "size": Array [
            "S",
            "L",
            "XL",
          ],
        },
        Object {
          "index": 1,
          "isExclusive": false,
          "isSale": false,
          "price": "$25.44",
          "productImage": "product-2.jpg",
          "productName": "Denim shirt",
          "size": Array [
            "XS",
            "S",
          ],
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "sc-htpNat kDFENX",
        "items": Array [
          Object {
            "index": 0,
            "isExclusive": false,
            "isSale": true,
            "price": "$18.88",
            "productImage": "product-1.jpg",
            "productName": "Striped shirt",
            "size": Array [
              "S",
              "L",
              "XL",
            ],
          },
          Object {
            "index": 1,
            "isExclusive": false,
            "isSale": false,
            "price": "$25.44",
            "productImage": "product-2.jpg",
            "productName": "Denim shirt",
            "size": Array [
              "XS",
              "S",
            ],
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
